packages/vite/src/node/plugins/esbuild.ts

```diff
@@ -200,192 +197,22 @@ function prettifyMessage(m: Message, code: string): string {
-   */
-  function isEscaped(jsonString: string, quotePosition: number) {
-    let index = quotePosition - 1
-    let backslashCount = 0
-
-    while (jsonString[index] === '\\') {
-      index -= 1
-      backslashCount += 1
-    }
-
-    return backslashCount % 2 === 1
-  }
-
-  let insideString = false
-  let offset = 0
-  let result = ''
-  let danglingCommaPos = null
-  for (let i = 0; i < jsonString.length; i++) {
-    const currentCharacter = jsonString[i]
-
-    if (currentCharacter === '"') {
-      const escaped = isEscaped(jsonString, i)
-      if (!escaped) {
-        insideString = !insideString
-      }
-    }
-
-    if (insideString) {
-      danglingCommaPos = null
-      continue
-    }
-    if (currentCharacter === ',') {
-      danglingCommaPos = i
-      continue
-    }
-    if (danglingCommaPos) {
-      if (currentCharacter === '}' || currentCharacter === ']') {
-        result += jsonString.slice(offset, danglingCommaPos) + ' '
-        offset = danglingCommaPos + 1
-        danglingCommaPos = null
-      } else if (!currentCharacter.match(/\s/)) {
-        danglingCommaPos = null
-      }
-    }
-  }
-
-  return result + jsonString.substring(offset)
-}
-
-async function readTSConfig(configPath: string): Promise<TSConfigJSON> {
-  const content: string = await new Promise((resolve, reject) => {
-    fs.readFile(configPath, 'utf-8', (err, data) => {
-      if (err) {
-        reject(err)
-        return
-      }
-      resolve(stripComments(stripBom(data)))
-    })
-  })
-
-  // tsconfig.json can be empty
-  if (/^\s*$/.test(content)) {
-    return {}
-  }
-
-  return JSON.parse(stripDanglingComma(content)) as TSConfigJSON
-}
-
-const tsconfigCache = new Map<string, TSConfigJSON>()
+const tsconfigCache = new Map<string, TSConfckParseResult>()
 async function loadTsconfigJsonForFile(
   filename: string
 ): Promise<TSConfigJSON> {
-  const directory = path.dirname(filename)
-
-  const cached = tsconfigCache.get(directory)
-  if (cached) {
-    return cached
-  }
-
-  let configPath = await findTSConfig(directory)
-  let tsconfig: TSConfigJSON = {}
-
-  if (configPath) {
-    const visited = new Set()
-    visited.add(configPath)
-
-    tsconfig = await readTSConfig(configPath)
-    while (tsconfig.extends) {
-      const configRequire = createRequire(configPath)
-
-      const extendsPath = configRequire.resolve(tsconfig.extends)
-      const extendedConfig = await readTSConfig(extendsPath)
-
-      if (visited.has(extendsPath)) {
-        throw new Error(
-          `Circular dependency detected in the "extends" field of ${configPath}`
-        )
-      }
-      visited.add(extendsPath)
-
-      tsconfig = {
-        extends: extendedConfig.extends,
-        compilerOptions: {
-          ...extendedConfig.compilerOptions,
-          ...tsconfig.compilerOptions
-        }
-      }
-      configPath = extendsPath
+  return parse(filename, {
+    cache: tsconfigCache,
+    resolveWithEmptyIfConfigNotFound: true
+  }).then(
```

Comment:
Now that it's inside an `async` function, plain `try { await ... } catch` seems more intuitive.

Reply:
uses try catch now